src/segment-loader.js

```diff
@@ -975,6 +976,28 @@ export default class SegmentLoader extends videojs.EventTarget {
     // equal to the last appended mediaIndex
     if (this.mediaIndex !== null) {
       this.mediaIndex -= mediaSequenceDiff;
+
+      // this can happen if we are going to load the first segment, but get a playlist
+      // update during that. mediaIndex would go from 0 to -1 if mediaSequence in the
+      // new playlist was incremented by 1.
+      if (this.mediaIndex < 0) {
+        this.mediaIndex = null;
+        this.partIndex = null;
+      } else {
+        const segment = this.playlist_.segments[this.mediaIndex];
+
+        // partIndex should remain the same for the same segment
+        // unless parts fell off of the playlist for this segment.
+        // In that case we need to reset partIndex and resync
+        if (this.partIndex && (!segment.parts || !segment.parts.length || !segment.parts[this.partIndex])) {
+          const mediaIndex = this.mediaIndex;
+
+          this.logger_(`currently processing part (index ${this.partIndex}) no longer exists.`);
+          this.resetLoader();
+
+          this.mediaIndex = mediaIndex;
```

Comment:
Is there a reason that we retain the media index after a reset?

Reply:
We want to throw away the partIndex and the data associated with it, but the mediaIndex is still valid. 
Example:
We are on mediaIndex 3 and partIndex 2, then a live refresh happens and we are on mediaIndex 2 and it no longer has parts.

Reply:
👍 would be worth a comment there